### PR TITLE
papis: update to 0.12.

### PR DESCRIPTION
**Pull request by icp1994**
#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, x86_64

### CODE REVIEW

File: srcpkgs/papis/template [hunk 1]
```diff
@@ -1,22 +1,34 @@
...
-pre_build() {
-	sed -i '/configparser/d' setup.py
+do_patch() {
+	vsed -e '/configparser/d' -i setup.py
+	vsed -e '/--cov=papis/d' -i setup.cfg
+}
+
+do_check() {
+	# Require to generate egg_info so that tests can use entry points
+	python3 setup.py egg_info
```
Comment (classabbyamp) [inline srcpkgs/papis/template:28]: this should be done in pre_check()

File: srcpkgs/papis/template [hunk 2]
```diff
@@ -1,22 +1,34 @@
...
 homepage="https://github.com/papis/papis"
+changelog="https://github.com/papis/papis/blob/master/CHANGELOG.md"
 distfiles="https://github.com/papis/papis/archive/v${version}.tar.gz"
-checksum=8ebb2f6f50e2e92e16a37ffc1a054983f6b50c6a56d3629d9f612f4197d7a7da
+checksum=75a91f827927a79ac648fbd810aacb1d2a0ba475f69f2f61c421c8b1dbf10cc9
 
-pre_build() {
-	sed -i '/configparser/d' setup.py
+do_patch() {
+	vsed -e '/configparser/d' -i setup.py
```
Comment (classabbyamp) [inline srcpkgs/papis/template:22]: why is this removed? It's also a single-line change, it would be better as a small patch.

File: srcpkgs/papis/template [hunk 3]
```diff
@@ -1,22 +1,34 @@
...
+	vsed -e '/--cov=papis/d' -i setup.cfg
```
Comment (classabbyamp) [inline srcpkgs/papis/template:23]: remove this and add `python3-pytest-cov` to `checkdepends`

File: srcpkgs/papis/template [hunk 4]
```diff
@@ -1,22 +1,34 @@
...
+}
+
+do_check() {
+	# Require to generate egg_info so that tests can use entry points
+	python3 setup.py egg_info
+
+	# Deselected one test which fails dring upstream CI too
+	# https://github.com/papis/papis/runs/6558190874?check_suite_focus=true
+	PYTHONPATH="$(cd build/lib* && pwd)" python3 -m pytest papis/ tests/ \
+		--deselect=tests/downloaders/test_project_euclid.py::test_1
```
Comment (classabbyamp) [inline srcpkgs/papis/template:33]: this can be turned into
```sh
make_check_pre="env PYTHONPATH=$wrksrc/build/lib"
make_check_args="--deselect=tests/downloaders/test_project_euclid.py::test_1"
make_check_target="papis tests"
```
and then defining do_check() is no longer needed

File: srcpkgs/papis/template [hunk 5]
```diff
@@ -1,22 +1,34 @@
...
- python3-colorama python3-click python3-slugify python3-prompt_toolkit
- python3-tqdm python3-Pygments python3-stevedore python3-parsing
- python3-filetype python3-bibtexparser python3-habanero python3-arxiv2bib
- python3-isbnlib python3 python3-setuptools python3-lxml
- python3-typing_extensions python3-doi"
+depends="python3-requests python3-filetype python3-parsing python3-arxiv2bib
+ python3-yaml python3-chardet python3-BeautifulSoup4 python3-colorama
+ python3-bibtexparser python3-click python3-habanero python3-isbnlib
+ python3-prompt_toolkit python3-tqdm python3-Pygments python3-stevedore
+ python3-doi python3-lxml python3-slugify python3-Whoosh"
```
Comment (classabbyamp) [inline srcpkgs/papis/template:11]: re-ordering all of these dependencies makes for a hard-to-parse diff, and shouldn't be necessary

Comment (icp1994) [reply srcpkgs/papis/template:11]: I was following the ordering in https://github.com/papis/papis/blob/v0.12/setup.py#L55 cause there are so many in case I miss something or something redundant gets into it.

File: srcpkgs/papis/template (hunk 2 above)
Comment (icp1994) [reply srcpkgs/papis/template:22]: is it okay to remove the `pre_build()`? it doesn't seem to be needed anymore

Comment (classabbyamp) [reply srcpkgs/papis/template:22]: yes, if it's empty

**Comment by icp1994** [comment]
Linter is complaining about `revision`, I don't know whether I was supposed to increment it in the forced push.

**Comment by classabbyamp** [comment]
it should not be, because you updated the version. Updating the revision should only be done if the PR as a whole changes something that isn't a version bump, not within the PR

File: srcpkgs/papis/template (hunk 5 above)
Comment (classabbyamp) [reply srcpkgs/papis/template]: I'd prefer not changing the order, it makes the history easier to use

**Comment by Goorzhel** [comment]
But if you're updating the package, as the diff shows, then you'll wanna set `revision=1`.

**Comment by icp1994** [comment]
Thanks for the info and reviews, updated properly.
